src/main/java/com/aws/greengrass/clientdevices/auth/certificate/CertificateStore.java

```diff
@@ -148,44 +147,46 @@ public X509Certificate getCACertificate() throws KeyStoreException {
         return certChain[0];
     }
 
-    private void setCaCertificateChain(Certificate... caCertificateChain) throws KeyStoreException {
-        if (caCertificateChain == null) {
-            throw new KeyStoreException("No certificate chain provided");
-        }
-        for (Certificate cert : caCertificateChain) {
-            if (!(cert instanceof X509Certificate)) {
-                throw new KeyStoreException("Unsupported certificate type");
-            }
-        }
-        setCaCertificateChain(Arrays.stream(caCertificateChain).toArray(X509Certificate[]::new));
-    }
 
     /**
-     * Sets the CA certificate chain.
+     * Sets the CA chain and private key that are used to generate certificates. It combines setting both values
+     * at the same time to avoid invalid states where the caChain can be updated without updating the value of
+     * the private key required to sign generated certificates.
+     *
+     * @param privateKey  leaf CA private key
+     * @param caCertificateChain a CA chain
      *
-     * @param caCertificateChain Array of CA certificates
-     * @throws KeyStoreException if unable to retrieve the certificate chain
+     * @throws KeyStoreException  if privateKey is not instance of PrivateKey or no ca chain provided
      */
-    public void setCaCertificateChain(X509Certificate... caCertificateChain) throws KeyStoreException {
+    public void setCA(Key privateKey, X509Certificate... caCertificateChain) throws KeyStoreException {
```

Comment:
I realized there was a potential for us to introduce a very sneaky bug in the scenario where we just call setCertificateChain and do not call setPrivateKey. It might lead to us signing certificates with an old private key. This change just changes the API so we have to set both at once in order to avoid making impossible states possible